File: app/Library/Models/ObjectModel.cs
namespace Library.Models
{
    public class Object
    {
        public int Id { get; set; }

        public DateTimeOffset CreatedDate { get; set; }
        public DateTimeOffset LastUpdateDate { get; set; }

        public Object()
        {
            CreatedDate = DateTimeOffset.Now;
            LastUpdateDate = DateTimeOffset.Now;
        }
    }
}

File: src/src/main/java/com/library/library/model/ObjectModel.java
package com.library.library.model;

import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Data
@SuperBuilder
@MappedSuperclass
@AllArgsConstructor
@NoArgsConstructor
public class ObjectModel {

    @Column(nullable = false, updatable = false)
    protected LocalDateTime createdDate;

    @Column(nullable = false)
    protected LocalDateTime lastUpdateDate;
}
